File: 2026-2/sesion-de-clase-4/dotnet/Lab01-2026-1/AssessmentViewerConsole.cs
namespace Lab01_2026_1;

public class AssessmentViewerConsole
{
    public static void Show(Assessment assessment)
    {
        Console.WriteLine("Inicio de examen ======");
        Console.WriteLine($"Duración: {assessment.Duracion}");

        DateTime fechaHora = assessment.FechaHora;
        Console.WriteLine($"Fecha Hora Inicio: {fechaHora:dd/MM/yyyy hh:mm}");

        IList<AssessmentItem> items = assessment.Items;
        int index = 0;
        int puntaje = 0;
        foreach (AssessmentItem item in items)
        {
            index++;
            Console.WriteLine($"Pregunta {index})");
            Question question = item.Pregunta;
            question.DevolverDatos();
            question.RecuperarRespuesta(item);
            bool esCorrecto = item.EsCorrecto;
            if (esCorrecto)
            {
                Console.WriteLine("Puntaje Pregunta: " + item.Puntaje);
                puntaje += item.Puntaje;
            }
            else
            {
                Console.WriteLine("Puntaje Pregunta: 0");
            }
        }
        assessment.Puntaje = puntaje;
        Console.WriteLine("Puntaje Final: " + puntaje);
    }
}

File: 2026-2/sesion-de-clase-4/java/Lab01-2026-1/pe/edu/pucp/lab/admin/AssessmentViewerConsole.java
package pe.edu.pucp.lab.admin;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

import pe.edu.pucp.lab.examenes.Assessment;
import pe.edu.pucp.lab.examenes.AssessmentItem;
import pe.edu.pucp.lab.preguntas.Question;

public class AssessmentViewerConsole {

    public static void show(Assessment assessment) {
        System.out.println("Inicio de examen ======");
        System.out.println(
                String.format("Duración: %d", assessment.getDuracion()));

        LocalDateTime fechaHora = assessment.getFechaHora();
        DateTimeFormatter formatter
                = DateTimeFormatter.ofPattern("dd/MM/yyyy hh:mm");
        System.out.println(
                String.format("Fecha Hora Inicio: %s",
                        fechaHora.format(formatter)));

        List<AssessmentItem> items = assessment.getItems();
        int index = 0;
        int puntaje = 0;
        for (AssessmentItem item : items) {
            System.out.println(
                    String.format("Pregunta %d)", ++index));
            Question question = item.getPregunta();
            question.devolverDatos();
            question.recuperarRespuesta(item);
            boolean esCorrecto = item.getEsCorrecto();
            if (esCorrecto) {
                System.out.println("Puntaje Pregunta: " + item.getPuntaje());
                puntaje += item.getPuntaje();
            } else {
                System.out.println("Puntaje Pregunta: 0");
            }
        }
        assessment.setPuntaje(puntaje);
        System.out.println("Puntaje Final: " + puntaje);
    }
}
